subsys/power/policy/policy_dummy.c

```diff
@@ -9,71 +9,33 @@
 #include <soc.h>
 #include "pm_policy.h"
 
-#define LOG_LEVEL CONFIG_PM_LOG_LEVEL /* From power module Kconfig */
 #include <logging/log.h>
-LOG_MODULE_DECLARE(power);
+LOG_MODULE_DECLARE(power, CONFIG_PM_LOG_LEVEL);
 
-#if !(defined(CONFIG_SYS_POWER_STATE_CPU_LPS_SUPPORTED) || \
-		defined(CONFIG_SYS_POWER_STATE_CPU_LPS_1_SUPPORTED) || \
-		defined(CONFIG_SYS_POWER_STATE_CPU_LPS_2_SUPPORTED) || \
-		defined(CONFIG_SYS_POWER_STATE_DEEP_SLEEP_SUPPORTED) || \
-		defined(CONFIG_SYS_POWER_STATE_DEEP_SLEEP_1_SUPPORTED) || \
-		defined(CONFIG_SYS_POWER_STATE_DEEP_SLEEP_2_SUPPORTED))
-#error "Enable Low Power States at SoC Level"
-#endif
-
-struct sys_pm_policy {
-	enum power_states pm_state;
-	int sys_state;
-};
-
-/* PM Policy based on SoC/Platform residency requirements */
-static struct sys_pm_policy pm_policy[] = {
-#ifdef CONFIG_SYS_POWER_STATE_CPU_LPS_SUPPORTED
-	{SYS_POWER_STATE_CPU_LPS, SYS_PM_LOW_POWER_STATE},
-#endif
-
-#ifdef CONFIG_SYS_POWER_STATE_CPU_LPS_1_SUPPORTED
-	{SYS_POWER_STATE_CPU_LPS_1, SYS_PM_LOW_POWER_STATE},
-#endif
-
-#ifdef CONFIG_SYS_POWER_STATE_CPU_LPS_2_SUPPORTED
-	{SYS_POWER_STATE_CPU_LPS_2, SYS_PM_LOW_POWER_STATE},
-#endif
-
-#ifdef CONFIG_SYS_POWER_STATE_DEEP_SLEEP_SUPPORTED
-	{SYS_POWER_STATE_DEEP_SLEEP, SYS_PM_DEEP_SLEEP},
-#endif
-
-#ifdef CONFIG_SYS_POWER_STATE_DEEP_SLEEP_1_SUPPORTED
-	{SYS_POWER_STATE_DEEP_SLEEP_1, SYS_PM_DEEP_SLEEP},
-#endif
-
-#ifdef CONFIG_SYS_POWER_STATE_DEEP_SLEEP_2_SUPPORTED
-	{SYS_POWER_STATE_DEEP_SLEEP_2, SYS_PM_DEEP_SLEEP},
-#endif
-};
-
-int sys_pm_policy_next_state(s32_t ticks, enum power_states *pm_state)
+enum power_states sys_pm_policy_next_state(s32_t ticks)
 {
-	static int cur_pm_idx;
-	int i = cur_pm_idx;
+	static u8_t cur_power_state;
+	u8_t i = cur_power_state;
```

Comment:
Why `u8_t`? Shouldn't it be `enum power_states`

Reply:
This is `u8_t` as we are doing mathematics operations on this variable later.

Reply:
In C arithmetic operations on enum are legal (but please correct me if I'm wrong). We also want to conform to Misra-C and we have this
```
	if (!sys_pm_ctrl_is_state_enabled(i)) {
```
with no cast. I am not sure any more what's the best way to handle this. In any case enum maps to `int`, not `u8_t`.

@ceolin could you have a quick look if this code conforms to Misra-C?

Reply:
Arithmetic on enum is possible in C as long as it stays in bounds of the enum. If we cross the boundary, the behaviour is unspecified. In C++ direct arithmetic on enum is not allowed.
I personally would like to keep the integers. If MISRA requires explicit casts, I can add them.

Reply:
this is not allowed, enums are a specific "essential type" in MISRA-C 2012 terminology and promote int (signed, unsigned, ...) are not allowed. We've to cast `i` here.

Reply:
this is not allowed, enums are a specific "essential type" in MISRA-C 2012 terminology and promote int (signed, unsigned, ...) are not allowed. We've to cast `i` here.

Reply:
Misra-C requires more careful type handling, explicit casting, etc. But I don't remember any more rules on enum to integer type conversion. We can fix it later, if required. Still, I believe we should change `u8_t` to `int`. Enums are stored as `int` (unless compiler option is forcing compact/short enums).

Reply:
> Misra-C requires more careful type handling, explicit casting, etc. But I don't remember any more rules on enum to integer type conversion. We can fix it later, if required. Still, I believe we should change `u8_t` to `int`. Enums are stored as `int` (unless compiler option is forcing compact/short enums).

this violates rule 10.3, a simple cast is enough

Reply:
@ceolin thanks for clarification!

Reply:
Added casts, changed type to int. I have left `static u8_t cur_power_state` to avoid wasting 3 bytes of ram.